source/sprite.c

```diff
@@ -2,39 +2,40 @@
 #include "util.h"
 #include "audio_utils.h"
 #include "soundbank.h"
+#include "pool.h"
 
 #include <tonc.h>
 #include <stdlib.h>
 #include <maxmod.h>
 
-#define MAX_SPRITES 128
 #define MAX_AFFINES 32
 #define SPRITE_FOCUS_RAISE_PX 10
 
 OBJ_ATTR obj_buffer[MAX_SPRITES];
 OBJ_AFFINE *obj_aff_buffer = (OBJ_AFFINE*)obj_buffer;
 
-static Sprite *free_sprites[MAX_SPRITES] = {NULL};
+typedef struct
+{
+    bool used;
+    Sprite* p_sprite;
+} _UsedSprite;
```

Comment:
So, this is a bit of an interesting change. I did need to readd this tracker, but mapping it directly to the array of sprite structs just isn't possible with the pool anymore. So, at the cost of space, there's this tracker.

I put a lot of thought into this, I may add more to this comment later.

But the TLDR;

- I don't want this to be a forced change to work with the pools, but I think there's value to this:
- We have control over the number of entries in the pool, as it's now decoupled from tracking it's OAM memory offset.
- Adding the OAM memory offset to the sprite struct is appropriate since it is strongly typed to the sprite itself.

Scrutiny is allowed.